src/WebJobs.Extensions.CosmosDB/Trigger/CosmosDBTriggerBinding.cs

```diff
@@ -48,14 +48,8 @@ public CosmosDBTriggerBinding(ParameterInfo parameter, DocumentCollectionInfo do
 
         public Task<ITriggerData> BindAsync(object value, ValueBindingContext context)
         {
-            IReadOnlyList<Document> triggerValue;
-            if (!TryAndConvertToDocumentList(value, out triggerValue))
-            {
-                throw new InvalidOperationException("Unable to convert trigger to CosmosDBTrigger.");
-            }
-
-            var valueBinder = new CosmosDBTriggerValueBinder(_parameter.ParameterType, triggerValue);
-            return Task.FromResult<ITriggerData>(new TriggerData(valueBinder, _emptyBindingData));
+            // ValueProvider is via binding rules. 
+            return Task.FromResult<ITriggerData>(new TriggerData(null, _emptyBindingData));
```

Comment:
So does this get called at all? Or is the value merged with this TriggerData somehow?

Reply:
It's still called to produce the binding data. But then we impose a  IValueProvider separately 

---
In reply to: [151226742](https://github.com/Azure/azure-webjobs-sdk-extensions/pull/331#discussion_r151226742) [](ancestors = 151226742)